### DISCUSSION TITLE
Imagemagick don't respect the memory limit with JXL effort 9

### DISCUSSION THREAD

**Discussion by MrJmpl3** [ImageMagick/ImageMagick#7410] [General]
I set the limit memory of Imagemagick to 2GB, but process use +10GB of RAM when tried to convert a big image with JXL effort 9, but when remove the JXL effort, the process works normally using 2GB. Any help?

**Comment by dlemstra** [comment]
The JXL library is an external dependency and I suspect that it needs this much memory when the JXL effort is set to 9. I don't think there is much that you can do about this.

**Reply by MrJmpl3** [reply]
Imagemagick can't limit memory usage?

**Reply by dlemstra** [reply]
There are options to limit the max allocations of the JXL library with the `max-memory-request` option but that will probably result in an error on the JXL side. I think that with effort 9 it just requires a lot more memory than without that setting.
